### DISCUSSION TITLE
Choose color scheme based on system's preferred theme

### DISCUSSION THREAD

**Discussion by spenserblack** [wezterm/wezterm#6297] [Q&A] [answered]
### What Operating System(s) are you running on?

Windows, Linux Wayland

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

I figured I'd start with a discussion question first to avoid making a duplicate issue (apparently I'm really bad at finding existing issues :sweat_smile:)

Is there a built-in way to specify a preferred dark and preferred light theme, and then have wezterm choose the correct theme based on system preferences? As someone who regularly switches my dark/light system preferences, this is a feature I've started appreciating for some tools.

I haven't seen anything about that in the config documentation, but just making sure I didn't miss anything :slightly_smiling_face: If it doesn't exist, I'm sure I can hack it together in the config file (I've done it before for Neovim).

**Comment by nekowinston** [comment, marked as answer]
See https://wezfurlong.org/wezterm/config/lua/wezterm.gui/get_appearance.html

**Reply by spenserblack** [reply]
Works perfectly! Thanks for the quick response!

```lua
local appearance_themes = {
  Light = light_theme,
  Dark = dark_theme,
}

local appearance = wezterm.gui.get_appearance()
config.color_scheme = appearance_themes[appearance] or dark_theme
```
